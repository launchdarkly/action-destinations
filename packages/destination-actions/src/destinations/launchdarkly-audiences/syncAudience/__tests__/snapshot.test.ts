import { createTestEvent, createTestIntegration } from '@segment/actions-core'
import { generateTestData } from '../../../../lib/test-data'
import destination from '../../index'
import nock from 'nock'

const testDestination = createTestIntegration(destination)
const actionSlug = 'syncAudience'
const destinationSlug = 'LaunchdarklyAudiences'
const seedName = `${destinationSlug}#${actionSlug}`

describe(`Testing snapshot for ${destinationSlug}'s ${actionSlug} destination action:`, () => {
  it('Audience entered', async () => {
    const action = destination.actions[actionSlug]
    const [eventData, settingsData] = generateTestData(seedName, destination, action, true)
    eventData['custom_audience_name'] = 'test_audience'
    eventData['segment_computation_action'] = 'audience'
    eventData['traits_or_props'] = { [eventData['custom_audience_name']]: true }

    nock(/.*/).persist().get(/.*/).reply(200)
    nock(/.*/).persist().post(/.*/).reply(200)
    nock(/.*/).persist().put(/.*/).reply(200)

    const event = createTestEvent({
      properties: eventData
    })

    const responses = await testDestination.testAction(actionSlug, {
      event: event,
      mapping: event.properties,
      settings: settingsData,
      auth: undefined
    })

    const request = responses[0].request
    const rawBody = await request.text()

    const json = JSON.parse(rawBody)
    expect(json).toMatchSnapshot()

    expect(request.headers).toMatchSnapshot()
  })

  it('Audience exited', async () => {
    const action = destination.actions[actionSlug]
    const [eventData, settingsData] = generateTestData(seedName, destination, action, true)
    eventData['custom_audience_name'] = 'test_audience'
    eventData['segment_computation_action'] = 'audience'
    eventData['traits_or_props'] = { [eventData['custom_audience_name']]: false }

    nock(/.*/).persist().get(/.*/).reply(200)
    nock(/.*/).persist().post(/.*/).reply(200)
    nock(/.*/).persist().put(/.*/).reply(200)

    const event = createTestEvent({
      properties: eventData
    })

    const responses = await testDestination.testAction(actionSlug, {
      event: event,
      mapping: event.properties,
      settings: settingsData,
      auth: undefined
    })

    const request = responses[0].request
    const rawBody = await request.text()

    const json = JSON.parse(rawBody)
    expect(json).toMatchSnapshot()

    expect(request.headers).toMatchSnapshot()
  })
})
