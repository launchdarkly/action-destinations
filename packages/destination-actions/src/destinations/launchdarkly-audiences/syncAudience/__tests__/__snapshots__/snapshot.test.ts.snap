// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing snapshot for LaunchdarklyAudiences's syncAudience destination action: Audience entered 1`] = `
Object {
  "batch": Array [
    Object {
      "cohortId": "test_audience",
      "cohortName": "Test audience",
      "userId": "Zzm(mpiX(u6tjxt",
      "value": true,
    },
  ],
  "environmentId": "Zzm(mpiX(u6tjxt",
}
`;

exports[`Testing snapshot for LaunchdarklyAudiences's syncAudience destination action: Audience entered 2`] = `
Headers {
  Symbol(map): Object {
    "authorization": Array [
      "Zzm(mpiX(u6tjxt",
    ],
    "content-type": Array [
      "application/json",
    ],
    "user-agent": Array [
      "SegmentSyncAudiences/1.0.0",
    ],
  },
}
`;

exports[`Testing snapshot for LaunchdarklyAudiences's syncAudience destination action: Audience exited 1`] = `
Object {
  "batch": Array [
    Object {
      "cohortId": "test_audience",
      "cohortName": "Test audience",
      "userId": "Zzm(mpiX(u6tjxt",
      "value": false,
    },
  ],
  "environmentId": "Zzm(mpiX(u6tjxt",
}
`;

exports[`Testing snapshot for LaunchdarklyAudiences's syncAudience destination action: Audience exited 2`] = `
Headers {
  Symbol(map): Object {
    "authorization": Array [
      "Zzm(mpiX(u6tjxt",
    ],
    "content-type": Array [
      "application/json",
    ],
    "user-agent": Array [
      "SegmentSyncAudiences/1.0.0",
    ],
  },
}
`;
